aip/general/0160.md

```diff
@@ -190,9 +190,25 @@ above what is defined here. For example, a service may support the logical
 operators but only permit a certain number of them (to avoid "queries of death"
 or other performance concerns).
 
-Further structure or limitations **must** be clearly documented, **must not**
-violate requirements set forth in this document, and a non-compliant filter
-query **must** error with `INVALID_ARGUMENT`.
+Further structure or limitations **must** be clearly documented, and
+**must not** violate requirements set forth in this document.
+
+### Validation
+
+If a non-compliant or schematically invalid `filter` string is specified,
+the API **should** error with `INVALID_ARGUMENT`. Wherever validation is relaxed
+for `filter`, the API **must** document the difference.
+
+Schematic validation refers, but is not limited to, the following:
+
+- Fields referenced in the `filter` **must** exist on the filtered schema
```

Comment:
Did we have a notion of a sort of "logical" field, or do we require the use of functions for that?

Reply:
can you explain that a bit more? Hwat do you mean by "logical" field?

Reply:
Something that may not end up being exposed as an actual field, but has some sort of logical value. For example, if we had an expiry timestamp, there could be some sort of "expires less than one minute from now" that you might want to filter on. I suspect that in most cases, output fields would be fine, and functions cover other situations, but it's just something to consider.

Reply:
Ah yes I see what you mean. In my mind I refer to those as "virtual" fields. I believe that is what functions are for atm.